### PR TITLE
shared io_service, /debug/close endpoint, stake wallets delayed loading

### PR DESCRIPTION
**Pull request by mbg033**

### CODE REVIEW

File: src/supernode/supernode.cpp [hunk 1]
```diff
@@ -198,6 +194,40 @@ void Supernode::setCoapRouters(ConnectionManager& coapcm)
     coapcm.addRouter(coap_router);
 }
 
+void Supernode::loadStakeWallets()
+{
+
+    // leak descriptors
+#if 0
```
Comment (bitkis) [inline src/supernode/supernode.cpp]: do we what to keep this code?

File: src/supernode/requests/debug.cpp [hunk 2]
```diff
@@ -119,13 +119,26 @@ Status doAnnounce(const Router::vars_t& vars, const graft::Input& input,
...
+}
+
 void __registerDebugRequests(Router &router)
 {
 #define _HANDLER(h) {nullptr, graft::supernode::request::debug::h, nullptr}
     // /debug/supernode_list/0 -> do not include inactive items
     // /debug/supernode_list/1 -> include inactive items
     router.addRoute("/debug/supernode_list/{all:[0-1]}", METHOD_GET, _HANDLER(getSupernodeList));
     router.addRoute("/debug/announce", METHOD_POST, _HANDLER(doAnnounce));
+    router.addRoute("/debug/close/", METHOD_POST, _HANDLER(closeStakeWallets));
```
Comment (bitkis) [inline src/supernode/requests/debug.cpp]: 'close_wallets'?

File: src/rta/fullsupernodelist.cpp [hunk 3]
```diff
@@ -386,6 +386,9 @@ void FullSupernodeList::selectTierSupernodes(const crypto::hash &block_hash, uin
 bool FullSupernodeList::loadWallet(const std::string &wallet_path)
 {
     bool result = false;
+//    if (this->size() >= 200)
```
Comment (bitkis) [inline src/rta/fullsupernodelist.cpp]: do we want to keep that?

File: src/supernode/supernode.cpp (hunk 1 above)
Comment (mbg033) [reply src/supernode/supernode.cpp]: not really, removed

File: src/supernode/requests/debug.cpp (hunk 2 above)
Comment (mbg033) [reply src/supernode/requests/debug.cpp]: thanks, sounds better, renamed to "close_wallets"

File: src/rta/fullsupernodelist.cpp (hunk 3 above)
Comment (mbg033) [reply src/rta/fullsupernodelist.cpp]: nope, removed

**Pull request merged**